portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,12 +41,26 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
+    String strNumComments = request.getParameter("numComments");
+    int numComments;
+    try{ 
+      numComments = Integer.parseInt(strNumComments);
+      if(numComments < 0) {
+          throw new Exception();
+      }
+    } catch (Exception e) { 
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
you should also return some kind of json error message here. This will help you debug in the future, since you'll get a bit more information than just "400 Bad Request".